katsdptelstate/rdb_reader.py

```diff
@@ -46,10 +53,23 @@ def zadd(self, key, score, member):
         self._zset[member] = score
 
     def end_sorted_set(self, key):
+        self.client_busy = True
         compat.zadd(self.client, key, self._zset)
+        self.client_busy = False
         self._zset = {}
 
 
+def _parse_rdb_file(parser, fd, filename=None):
+    """Apply RDB parser to file descriptor, raising RdbParseError on error."""
+    try:
+        parser.parse_fd(fd)
+    except Exception as exc:
+        # Don't remap exception to RdbParseError if it originates from callback
+        if parser._callback.client_busy:
```

Comment:
`_callback` is private. Rather pass the callback as an extra parameter to this function.

Reply:
Done.